tests/verify_riak_stats.erl

```diff
@@ -182,10 +197,35 @@ compare_http_and_console_stats(Stats1, Stats2) ->
 			not lists:keymember(K, 1, Stats1)],
     lager:info("OnlyInHttp = ~p~n"
 	       "OnlyInAdmin = ~p~n", [OnlyInHttp, OnlyInAdmin]),
+    maybe_log_stats_keys("Keys missing from riak-admin", OnlyInHttp),
+    maybe_log_stats_keys("Keys missing from HTTP", OnlyInAdmin),
     ?assertEqual([], OnlyInHttp),
     ?assertEqual([], OnlyInAdmin),
     ok.
 
+verify_stats_keys_complete(Node, Stats) ->
+    ActualKeys = proplists:get_keys(Stats),
+    ExpectedKeys = all_stats(Node),
+    MissingStatsKeys = diff_lists(ActualKeys, ExpectedKeys),
+    AdditionalStatsKeys = diff_lists(ExpectedKeys, ActualKeys),
+    maybe_log_stats_keys(MissingStatsKeys, "missing stats keys"),
+    maybe_log_stats_keys(AdditionalStatsKeys, "additional stats"),
+    ?assert(length(MissingStatsKeys) == 0),
+    ?assert(length(AdditionalStatsKeys) == 0),
+    ok.
+
+diff_lists(List, ThatList) ->
+    lists:filter(fun(Element) -> not lists:member(Element, List) end, ThatList).
+
+maybe_log_stats_keys(StatsKeys, _Description) when length(StatsKeys) == 0 ->
+    ok;
+maybe_log_stats_keys(StatsKeys, Description) ->
+    lager:info("~s: ~s", [Description, pretty_print_stats_keys(StatsKeys)]).
+
+pretty_print_stats_keys(StatsKeys) ->
+    ConvertedStatsKeys = lists:map(fun(Stat) -> binary_to_list(Stat) end, StatsKeys),
```

Comment:
I get a `badarg` consistently coming from this line. Are you sure the inputs are all binaries?

```
================ verify_riak_stats failure stack trace =====================
{badarg,[{erlang,binary_to_list,"K",[]},
         {verify_riak_stats,'-pretty_print_stats_keys/1-fun-0-',1,
                            [{file,"tests/verify_riak_stats.erl"},{line,226}]},
         {lists,map,2,[{file,"lists.erl"},{line,1224}]},
         {verify_riak_stats,pretty_print_stats_keys,1,
                            [{file,"tests/verify_riak_stats.erl"},{line,226}]},
         {verify_riak_stats,maybe_log_stats_keys,2,
                            [{file,"tests/verify_riak_stats.erl"},{line,223}]},
         {verify_riak_stats,compare_http_and_console_stats,2,
                            [{file,"tests/verify_riak_stats.erl"},{line,200}]},
         {verify_riak_stats,confirm,0,
                            [{file,"tests/verify_riak_stats.erl"},{line,49}]},
         {riak_test_runner,return_to_exit,3,
                           [{file,"src/riak_test_runner.erl"},{line,159}]}]}
============================================================================
```